### PR TITLE
Typesafe Queues

### PR DESCRIPTION
**Pull request by makosblade**
This PR:
- removes `SIGNAL` queue as it is unused in the codebase
- Adds a typesafe queue wrapper

### CODE REVIEW

File: src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    fun getQueue(hazelcast: HazelcastInstance): IQueue<T> {
+        this.checker.check()
+        return hazelcast.getQueue(name)
+    }
+
+    companion object {
+        private val instanceChecker = UniqueInstanceManager(HazelcastQueue::class.java)
+
+        // When adding new entries to this list, please make sure to keep it sorted and keep the name in sync
+        @JvmField val AUDITING = HazelcastQueue<AuditableEvent>("AUDITING")
```
Comment (dbt) [inline src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: `AUDITING` and `EMAIL_SPOOL` don't have `QueueConfigurer`s, is that bad?

File: src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        @JvmField val BACKGROUND_INDEXING  = HazelcastQueue<EntitySet>("BACKGROUND_INDEXING")
```
Comment (dbt) [inline src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: Also currently unused?

Comment (makosblade) [reply src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: Yep, this is used in a future PR

File: src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt (hunk 1 above)
Comment (makosblade) [reply src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: 🤔 It's not ~great~ since they won't have a max size set I think. I can add that

Comment (you-leee) [reply src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: They will have, because of the default we set up: `10_000`

Comment (dbt) [reply src/main/kotlin/com/openlattice/hazelcast/HazelcastQueue.kt]: They will have a default size but also a default TTL of 60s.

**Pull request merged**